src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs

```diff
@@ -1429,16 +1450,79 @@ FROM aggregates
             .ToDictionary(g => g.Key, g => g.ToArray());
     }
 
+    private async Task<List<UnverifiedTwoWayLinkEntryHistory>> GetUnverifiedTwoWayLinks(ICollection<Entity> entities, GatewayModel.LedgerState ledgerState, CancellationToken token = default)
```

Comment:
This took a while to understand - this is actually a collection of both:
* All unverified links E=>E1 for each given entity E
* AND all unverified links E1=>E2 for each given first-linked E2

In the general case, this feels very expensive. We could have O(P * N * N) here for `P` = page size and `N` = all unverified links.

I think it would be more performant to verify each link type explicitly in the query? And we can also avoid lots of the `TwoWayLinkResolver` stuff.
e.g. dAppDefinition link E => D requires checking that (A) D is a dapp definition and (B) D has a dAppEntity link back.

Reply:
As I wrote earlier this is a naive implementation and I'll almost certainly change it to only return entries potentially useful for two-way link verification, thus eliminating this `N * N` from the complexity.

Reply:
Changed to SQL-based implementation.